solidity/contracts/UpdateStorageMirrorGuard.sol

```diff
@@ -52,14 +52,16 @@ contract UpdateStorageMirrorGuard is BaseGuard {
     bytes memory _signatures,
     address _msgSender
   ) external {
-    didSettingsChange = true;
+    didSettingsChangeForSafe[_msgSender] = true;
     // TODO: change these data with the decoded ones
     address[] memory _owners = new address[](1);
+    _owners[0] = _msgSender;
     IStorageMirror.SafeSettings memory _safeSettings = IStorageMirror.SafeSettings({owners: _owners, threshold: 1});
+    safeSettings[_msgSender] = _safeSettings;
 
-    settingsHash = keccak256(abi.encode(_safeSettings));
+    bytes32 _settingsHash = keccak256(abi.encode(_safeSettings));
 
-    emit SettingsChanged(msg.sender, settingsHash, _safeSettings);
+    emit SettingsChanged(msg.sender, _settingsHash, _safeSettings);
```

Comment:
Since we are using plaintex now i hink we should emit tthis at the end of the post-hook